nexus/internal/uploader/uploader_default.go

```diff
@@ -42,24 +45,42 @@ func (u *DefaultUploader) Upload(task *UploadTask) error {
 		}
 	}(file)
 
-	req, err := retryablehttp.NewRequest(
-		http.MethodPut,
-		task.Url,
-		file,
-	)
-
-	for _, header := range task.Headers {
-		parts := strings.Split(header, ":")
-		req.Header.Set(parts[0], parts[1])
+	fileWithLen, err := NewFileWithLen(file)
+	if err != nil {
+		return err
 	}
-
+	req, err := retryablehttp.NewRequest(http.MethodPut, task.Url, fileWithLen)
 	if err != nil {
 		return err
 	}
+	for _, header := range task.Headers {
+		parts := strings.Split(header, ":")
+		req.Header.Set(parts[0], parts[1])
+	}
 
 	if _, err = u.client.Do(req); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+type FileWithLen struct {
+	*os.File
+	len int
+}
+
+func NewFileWithLen(file *os.File) (FileWithLen, error) {
+	stat, err := file.Stat()
+	if err != nil {
+		return FileWithLen{}, err
+	}
+	if stat.Size() > math.MaxInt {
+		return FileWithLen{}, fmt.Errorf("file larger than %v", math.MaxInt)
```

Comment:
hmm should we be more conservative here and log a warning and return 0 size? In my testing, content-length is not strictly required for certain cloud vendors so we may still be able to upload the file successfully here